stubs/Pygments/pygments/formatter.pyi

```diff
@@ -10,6 +12,11 @@ class Formatter:
     title: Any
     encoding: Any
     options: Any
-    def __init__(self, **options) -> None: ...
+    @overload
+    def __init__(self: Formatter[str], *, encoding: None = ..., **options) -> None: ...
+    @overload
+    def __init__(self: Formatter[bytes], *, encoding: str, **options) -> None: ...
+    @overload
+    def __init__(self: Formatter[bytes], *, outencoding: str, **options) -> None: ...
```

Comment:
All overloads should specify what happens with both `encoding` and `outencoding`:

```suggestion
    @overload
    def __init__(self: Formatter[str], *, encoding: None = ..., outencoding: None = ..., **options) -> None: ...
    @overload
    def __init__(self: Formatter[bytes], *, encoding: str, outencoding: None = ..., **options) -> None: ...
    @overload
    def __init__(self: Formatter[bytes], *, encoding: None = ..., outencoding: str, **options) -> None: ...
```

Reply:
Turns out it's possible to provide both, but we probably shouldn't allow that. Doing that is most likely a mistake.

```
>>> from pygments.formatter import Formatter
>>> print(Formatter(encoding='x', outencoding='y').encoding)
y
```

Reply:
Yeah, that's why I just allowed the `**options` to take up the other parameter, but I can add this change if preferred.

Reply:
I think it's better to be explicit about this. There is an open mypy PR that changes which overload is chosen in these situations: python/mypy#11589